-1 
Tested at: d04eea3320085461cbabcaba94ea20659a3c2185
 I found an error when building: 
 
 <pre>Copying tmp/slc6_amd64_gcc493/src/CondCore/Utilities/bin/conddb_test_gt_perf/conddb_test_gt_perf to productstore area:
Copying tmp/slc6_amd64_gcc493/src/CondCore/Utilities/bin/cmscond_export_database/cmscond_export_database to productstore area:
>> Building binary conddb_test
>> Building binary conddb_import
Copying tmp/slc6_amd64_gcc493/src/CondCore/Utilities/bin/cmscond_create_key/cmscond_create_key to productstore area:
error: class 'reco::CSCHaloData' has a different checksum for ClassVersion 10. Increment ClassVersion to 11 and assign it to checksum 1723373351
>> Building binary cmscond_encode_db_file
Suggestion: You can run 'scram build updateclassversion' to generate src/DataFormats/METReco/src/classes_def.xml.generated with updated ClassVersion
gmake: *** [tmp/slc6_amd64_gcc493/src/DataFormats/METReco/src/DataFormatsMETReco/libDataFormatsMETRecoCapabilities.so] Error 1
>> Building binary conddb_copy_iov
Copying tmp/slc6_amd64_gcc493/src/CondCore/Utilities/bin/cmscond_export_iov/cmscond_export_iov to productstore area:
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11266/8106/summary.html 

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/833f2d72b06252ebcfa9010d06779fc6fc96d661
https://github.com/cms-sw/cmssw/commit/608fea7433207a9f57fdcacc4b410d9af04f0a2f
https://github.com/cms-sw/cmssw/commit/0a6bc35402c18834d754891f829f597183f03ad7
https://github.com/cms-sw/cmssw/commit/e3ad181dff149dd2c11ebd5c6103fb45f7c1c1e1
https://github.com/cms-sw/cmssw/commit/cdb5396afcf4bf9a8fca372bd2763c3608fdc00d
https://github.com/cms-sw/cmssw/commit/cdb62e7c34bc5727c2246e89f4d24f6cfeeeea53
https://github.com/cms-sw/cmssw/commit/6e3888f7d40aeed632dc840f696f9e2ca0391be3
https://github.com/cms-sw/cmssw/commit/c72f07dbcdc9e340af2a3724b2f27f300d5cb12e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11266/8106/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11266/8106/git-merge-result

--- a/DataFormats/METReco/src/classes_def.xml
+++ b/DataFormats/METReco/src/classes_def.xml
@@ -175,7 +175,8 @@
   <class name="std::vector<HaloTowerStrip>"/>
   <class name="edm::Wrapper<std::vector<HaloTowerStrip> >"/>
 
-  <class name="reco::CSCHaloData" ClassVersion="10">
+  <class name="reco::CSCHaloData" ClassVersion="11">
+   <version ClassVersion="11" checksum="1723373351"/>
    <version ClassVersion="10" checksum="3522217120"/>
   </class>
   <class name="edm::Wrapper<reco::CSCHaloData>"/>

--- a/RecoMET/METAlgorithms/src/CSCHaloAlgo.cc
+++ b/RecoMET/METAlgorithms/src/CSCHaloAlgo.cc
@@ -651,7 +651,7 @@ reco::CSCHaloData CSCHaloAlgo::Calculate(const CSCGeometry& TheCSCGeometry,
 	 float jZ = jGlobalPosition.z() ;
 	 float jT = jSegment->time();
 	 //	 if(abs(jZ)<650&& TheEvent.id().run() < 251737)jT-= 25;
-	 if (TMath::ACos(TMath::Cos(jPhi - iPhi)) <= 0.2//max_segment_phi_diff 
+	 if ( abs(deltaPhi(jPhi , iPhi)) <= 0.2//max_segment_phi_diff 
 	     //&& abs(jR - iR) <= max_segment_r_diff 
 	     && (abs(jR - iR) <= max_segment_r_diff || (abs(jR - iR)<0.03*abs(jZ - iZ) &&  jZ*iZ<0) )
 	     && (jTheta < max_segment_theta || jTheta > TMath::Pi() - max_segment_theta)) {

